wallet/src/repository/wallet/mod.rs

```diff
@@ -313,23 +319,22 @@ where
         // Query database `transaction_next_id` to compute total amount of transactions
         let db_total = self
             .db
-            .get_or_default::<_, u32>(&keys::transaction_next_id(account))?
-            .saturating_sub(1);
+            .get_or_default::<_, u32>(&keys::transaction_next_id(account))?;
 
         if db_total > 0 {
```

Comment:
```suggestion
```

You could remove this if because when `db_total == 0` the only possible range is `0..0`, and that's a valid range with 0 elements.